sgv2-docsapi/src/main/java/io/stargate/sgv2/docsapi/service/json/JsonConverter.java

```diff
@@ -0,0 +1,383 @@
+  }
+
+  /**
+   * Adds an Object's paths to the DeadLeafCollector.
+   *
+   * @param ref the Object to mark for deletion.
+   * @param parentPath the parent path to mark for deletion.
+   * @param collector the DeadLeafCollector
+   */
+  private void markObjectAtPathAsDead(
+      JsonNode ref, String parentPath, DeadLeafCollector collector) {
+    if (!ref.isObject()) { // it's a scalar
+      collector.addLeaf(parentPath, ImmutableDeadLeaf.builder().name("").build());
+    } else {
+      Iterator<String> fieldNames = ref.fieldNames();
+      while (fieldNames.hasNext()) {
+        String fieldName = fieldNames.next();
+        collector.addLeaf(parentPath, ImmutableDeadLeaf.builder().name(fieldName).build());
+      }
+    }
+  }
+
+  /**
+   * Adds an Array's paths to the DeadLeafCollector.
+   *
+   * @param ref the Array to mark for deletion.
+   * @param parentPath the parent path to mark for deletion.
+   * @param collector the DeadLeafCollector
+   */
+  private void markArrayAtPathAsDead(JsonNode ref, String parentPath, DeadLeafCollector collector) {
+    if (!ref.isArray()) { // it's a scalar
+      collector.addLeaf(parentPath, ImmutableDeadLeaf.builder().name("").build());
+    } else {
+      collector.addArray(parentPath);
+    }
+  }
+
+  /**
+   * Writes leaf data to the JsonNode in place, but only if the writetime of the leaf is newer than
+   * the current data at the path in the JsonNode.
+   *
+   * @param ref the JsonNode
+   * @param row the Row, for extraction of values
+   * @param leaf the path of the leaf
+   * @param parentPath the parent path to the leaf in the document
+   * @param pathWriteTimes a Map of all the path write times
+   * @param rowWriteTime the current row's write time
+   * @param numericBooleans whether to treat boolean values should be treated as integers
+   */
+  private void writeLeafIfNewer(
+      JsonNode ref,
+      RowWrapper row,
+      String leaf,
+      String parentPath,
+      Map<String, Long> pathWriteTimes,
+      Long rowWriteTime,
+      boolean numericBooleans) {
+    JsonNode n = NullNode.getInstance();
+
+    if (!row.isNull(docsProperties.tableProperties().stringValueColumnName())) {
+      String value = row.getString(docsProperties.tableProperties().stringValueColumnName());
+      if (value.equals(Constants.EMPTY_OBJECT_MARKER)) {
+        n = mapper.createObjectNode();
+      } else if (value.equals(Constants.EMPTY_ARRAY_MARKER)) {
+        n = mapper.createArrayNode();
+      } else {
+        n = new TextNode(value);
+      }
+    } else if (!row.isNull(docsProperties.tableProperties().booleanValueColumnName())) {
+      Boolean booleanFromRow;
+      if (numericBooleans) {
+        booleanFromRow =
+            (row.getByte(docsProperties.tableProperties().booleanValueColumnName()) != 0);
+      } else {
+        booleanFromRow = row.getBoolean(docsProperties.tableProperties().booleanValueColumnName());
+      }
+
+      n = BooleanNode.valueOf(booleanFromRow);
```

Comment:
So what's the reason not to go for:

```
DocsApiUtils#getBooleanFromRow(row, documentProperties, numericBooleans)
```

This way we reuse the same logic everywhere.. In fact, it would be way easier to read values before if and switch:

```
String stringValue = DocsApiUtils.getStringFromRow(row, documentProperties);
Boolean booleanValue = DocsApiUtils.getBooleanFromRow(row, documentProperties, numericBooleans);
String doubleValue = DocsApiUtils.getDoubleFromRow(row, documentProperties);

if (stringValue != null) {
   ...
} else if (booleanValue != null) {
  ...

} else if (doubleValue != null) {
  ...
}
```

Reply:
Ah yea, I should've done this - my bad. idk how I didn't notice to do this

Will fix this